src/_pytest/main.py

```diff
@@ -367,6 +365,22 @@ def pytest_runtestloop(session: Session) -> bool:
     return True
 
 
+def _in_build(path: Path) -> bool:
+    """Attempt to detect if ``path`` is the root of a buildsystem's artifacts
+    by checking known dirnames patterns, and the presence of configuration in
+    the parent dir by checking for a setup.py, setup.cfg, or pyproject.toml.
+    """
+    if not path.is_dir():
+        return False
+
+    if any(fnmatch_ex(pat, path) for pat in ("build", "dist")):
+        indicators = ("setup.py", "setup.cfg", "pyproject.toml")
+        if any((path.parent / f).is_file() for f in indicators):
```

Comment:
It feels like this is cheaper than fnmatch. Could you try adding a “guard expression” and bailing early if none of these exist?